changelogs/fragments/30413-lineinfile-concurrence_issue.yaml

```diff
@@ -0,0 +1,3 @@
+bugfixes:
+- "Fixed lineinfile concurrence issue with multiple delegation on same delegated host and same file"
+- "common.file.FileLocking now properly handles lock_timeout = None"
```

Comment:
Change this entry to state that "`common.file.FileLock.set_lock()` now handles negative values the same as zero". The `set_lock()` method seems to already handle `None` properly, but not negative values.

Reply:
Hmm, actually, I'm not sure this change is correct.

Reply:
Will do!
Actually the lock_timeout <= 0 bailed out when None, however now it handles both negative and None, whereas before it behaved a bit unexpected. I'll set up some integration tests as soon as possible!